test_ec2_metadata.py

```diff
@@ -138,3 +138,22 @@ def test_region(resps):
 def test_reservation_id(resps):
     add_response(resps, 'reservation-id', 'r-12345678901234567')
     assert ec2_metadata.reservation_id == 'r-12345678901234567'
+
+
+def test_security_groups(resps):
+    # most common case: a single SG
+    add_response(resps, 'security-groups', 'security-group-one')
+    assert ec2_metadata.security_groups == ['security-group-one']
```

Comment:
please split the tests into one function per test, and then merge the comments into the test function names, e.g. `test_security_groups_one` , `test_security_groups_many`, `test_security_groups_empty_string`. Tests should follow the AAA pattern ( http://jamescooke.info/arrange-act-assert-pattern-for-python-developers.html ). It bulk-heads the tests so a failure in one assertion doesn't prevent the others being run, and it avoids state building up, which you're fixing here with `clear_all()` being repeatedly invoked in the tests but there's no need since we already automatically do that between tests 👌